### PR TITLE
chore: enable sorting in semantic viewer

### PR DESCRIPTION
**Pull request by magnew**
Closes: #11033 

### Description:

Enables sorting in the semantic viewer. Users can:
- Sort by one column at a time
- Change sort direction

What's left:
- Sort UI/style obviously needs work (#11350) 
- Sorting on pivoted results needs more thought. You can sort, but the behavior within groups isn't right, everything gets sorted. We tried to understand the correct behavior from the explores page, but it's not clear exactly what it's doing there either. Additionally, the current dataset in the semantic viewer isn't great for testing pivoted sorting. I'm suggesting we merge this as incremental progress and I have filed a ticket to improve pivoted sorting: #11349 

Kapture 2024-09-02 at 18 57 58

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/SemanticLayerService/Pivoting.ts [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+    SemanticLayerPivot,
+    SemanticLayerResultRow,
+    SemanticLayerSortBy,
+} from '@lightdash/common';
 import pl from 'nodejs-polars';
 
 export function pivotResults(
     results: SemanticLayerResultRow[],
     { values, ...options }: SemanticLayerPivot,
+    sortBy?: SemanticLayerSortBy[],
```
Comment (almeidabbm) [inline packages/backend/src/services/SemanticLayerService/Pivoting.ts:11]: Can we remove this as it is not being used? 🙇

File: packages/backend/src/services/SemanticLayerService/SemanticLayerService.ts [hunk 2]
```diff
@@ -212,7 +212,7 @@ export class SemanticLayerService extends BaseService {
             const pivotedResults =
                 query.pivot.index.length === 0
                     ? results
-                    : pivotResults(results, query.pivot);
+                    : pivotResults(results, query.pivot, query.sortBy);
```
Comment (almeidabbm) [inline packages/backend/src/services/SemanticLayerService/SemanticLayerService.ts:215]: same here

**Review by almeidabbm — APPROVED** [review]
LGTM 🚀

File: packages/backend/src/services/SemanticLayerService/Pivoting.ts (hunk 1 above)
Comment (magnew) [reply packages/backend/src/services/SemanticLayerService/Pivoting.ts:11]: Yup. Wasn't sure for a bit there if we needed to keep it. Will take it out.

**Pull request merged**